### PR TITLE
Migrate to REST API v2

### PR DESCRIPTION
**Pull request by lefcha**

### CODE REVIEW

File: tests/test_api_comments.py [hunk 1]
```diff
@@ -1,20 +1,11 @@
...
 import responses
 
-from tests.data.test_defaults import (
-    DEFAULT_REQUEST_ID,
-    INVALID_ENTITY_ID,
-    REST_API_BASE_URL,
-)
-from tests.utils.test_utils import (
-    assert_auth_header,
-    assert_id_validation,
```
Comment (PotHix) [inline tests/test_api_comments.py:15]: I can see this being removed everywhere, but not from test_utils. Is this still needed?

File: todoist_api_python/authentication.py [hunk 2]
```diff
@@ -43,9 +45,8 @@ def revoke_auth_token(
         "client_secret": client_secret,
         "access_token": token,
     }
-    response = post(session=session, url=endpoint, data=payload)
-
-    return response
+    post(session=session, url=endpoint, data=payload)
+    return True
```
Comment (PotHix) [inline todoist_api_python/authentication.py:49]: I don't see any changes to this. Why are we ignoring the response and always returning True for this and other endpoints? Are we sure this never fails?

**Review by PotHix — APPROVED** [review]
A few minor comments, but LGTM.

File: tests/test_api_comments.py (hunk 1 above)
Comment (lefcha) [reply tests/test_api_comments.py:15]: Thanks, forgot to remove this. Resolved by: https://github.com/Doist/todoist-api-python/pull/61/commits/aee3b29e862da7fa50b270bd5392162deed48190

File: todoist_api_python/authentication.py (hunk 2 above)
Comment (lefcha) [reply todoist_api_python/authentication.py]: I think this was accidental, while I was testing something, or trying to fix some pyright error. Good catch. Resolved by: https://github.com/Doist/todoist-api-python/pull/61/commits/6f448d22df9ed370e83e4b14a95f65be757c9c92

Comment (PotHix) [reply todoist_api_python/authentication.py]: This is happening for most methond on [todoist_api_python/api.py](https://github.com/Doist/todoist-api-python/pull/61/files#diff-c9e0cccf5b7f01426dcdc94fe12598c1b8f1629de0a2014472479aee4fcbc702). Are you sure this is accidental?

Comment (lefcha) [reply todoist_api_python/authentication.py]: Hm, let me have a closer look, I experimented with the return type of [todoist_api_python/http_requests.py](https://github.com/Doist/todoist-api-python/blob/6f448d22df9ed370e83e4b14a95f65be757c9c92/todoist_api_python/http_requests.py), but I don't think I modified the return types of the API functions. This is weird.  Let me fix it.

Comment (lefcha) [reply todoist_api_python/authentication.py]: Should be resolved by https://github.com/Doist/todoist-api-python/pull/61/commits/21d8caff54e866f7351d1c489ee0ce190a0de543

Comment (PotHix) [reply todoist_api_python/authentication.py]: Thanks! Looks good now. I'm closing this comment and ask for more changes in a different one.

**Comment by PotHix** [comment]
@lefcha we will have to adapt the client to return the object on all update methods: https://github.com/Doist/Todoist/pull/6938

This is the last change, I think. :)

**Comment by lefcha** [comment]
> @lefcha we will have to adapt the client to return the object on all update methods: [Doist/Todoist#6938](https://github.com/Doist/Todoist/pull/6938)
> 
> This is the last change, I think. :)

Will the backend return `200` instead of `204`?

Because in that case the object should be returned, according to:

https://github.com/Doist/todoist-api-python/blob/21d8caff54e866f7351d1c489ee0ce190a0de543/todoist_api_python/http_requests.py#L44-L48

**Comment by PotHix** [comment]
> Will the backend return 200 instead of 204?

The update will return 200 with the JSON of the updated object.

**Pull request merged**